[Joshua Nee] iP

dZhei98 opened the pull request.

dZhei98 committed:
Add greet functionality

Level-0 Greet

dZhei98 committed:
Update coding style for constants

Update horizontal_line, introduction, exit

dZhei98 committed:
Add echo function

dZhei98 committed:
Add functionality to store input

Add the ability to store whatever text entered by the user and display them back to the user when requested

dZhei98 committed:
Add Task and TaskManager class

Add the ability to mark tasks as done
Add the ability to change the status back to not done

dZhei98 committed:
Add comments

Check coding standard

dZhei98 committed:
Shift taskCount from TaskManager to Task

dZhei98 committed:
Add support for tracking ToDos, Events, Deadlines

Add Todo class
Add Deadline class
Add Event class
Add Parser class
Update toString for Task class

dZhei98 committed:
Add test cases

dZhei98 committed:
Add InputManager and update Parser

Add parseTaskNumber method in Parser
Set phrases to public

P0tatoChips commented on src/main/java/InputManager.java:
```diff
@@ -0,0 +1,48 @@
+import java.util.Scanner;
+
+public class InputManager {
+    public static final String EXIT_PHRASE = "bye";
+    public static final String LIST_PHRASE = "list";
+    public static final String MARK_PHRASE = "mark";
+    public static final String UNMARK_PHRASE = "unmark";
+    public static final String TODO_PHRASE = "todo";
+    public static final String DEADLINE_PHRASE = "deadline";
+    public static final String BY_PHRASE = "/by";
+    public static final String EVENT_PHRASE = "event";
+    public static final String AT_PHRASE = "/at";
```
I like how you followed the naming conventions.

P0tatoChips commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,43 @@
+public class TaskManager {
+    private static final int MAX_NUM_OF_TASKS = 100;
+    private static Task[] tasks = new Task[MAX_NUM_OF_TASKS];
```
I liked how you did this instead of using magic numbers.

P0tatoChips commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,78 @@
+
+            newTask = new Event(description, at);
+            break;
+
```
The rest of the cases don't have newline after the break; statement, maybe you can remove the newline to standardize?

dZhei98 replied:
Thanks for pointing this out! It has been addressed

P0tatoChips commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,28 @@
+
+    private static void printIntroduction() {
+        final String INTRODUCTION =
+                "Hihi, my name is Jay!" + System.lineSeparator() + "What can I do for you today?";
```
I liked how you got creative with the greetings.

P0tatoChips reviewed (commented):
Overall, great code that generally followed the coding convections and coding standard. 👍

markusteim commented on src/main/java/InputManager.java:
```diff
@@ -0,0 +1,48 @@
+import java.util.Scanner;
+
+public class InputManager {
+    public static final String EXIT_PHRASE = "bye";
```
I like the big amount of specific and clarified constants

markusteim commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,78 @@
+
+            newTask = new Event(description, at);
+            break;
+
```
You could remove the newline to keep the overall code similar.

dZhei98 replied:
Thanks for pointing this out! It has been addressed

markusteim commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,78 @@
+
+            // add one to remove space
+            description = input.substring(descriptionIndex + InputManager.TODO_PHRASE.length() + 1,
+                    input.length());
```
I like how you have broken the lines up to make it more readable

markusteim commented on src/main/java/InputManager.java:
```diff
@@ -0,0 +1,48 @@
+                System.out.println(Duke.HORIZONTAL_LINE);
+
+                // handle list, mark, unmark, add
+                if (input.equals(LIST_PHRASE)) {
```
I like how you have kept it nice and clean here

markusteim reviewed (commented):
Overall nice and good code following standards

bdthanh commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,43 @@
+
+    public static void printList() {
+        for (int i = 0; i < Task.getTaskCount(); i += 1) {
+            System.out.println(Integer.toString(i + 1) + ". " + tasks[i]);
```
No need Integer.toString to convert (i + 1) to string since println() will do it for you

bdthanh commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,78 @@
+
+            newTask = new Deadline(description, by);
+            break;
+        case InputManager.EVENT_PHRASE:
```
Personally, I prefer a short code for switch statements. You can create different function to handle each case. But I think it should be alright

bdthanh commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    public static void markTask(String input) {
+        int taskNum = Parser.parseTaskNumber(InputManager.MARK_PHRASE, input);
```
I think no need InputManager.MARK_PHRASE, MARK_PHRASE only works oso

bdthanh commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,38 @@
+list
+unmark 6
+list
+bye
```
good number of test commands

bdthanh reviewed (commented):
Overall good job! No coding standard violation, perfect naming for constant, variables and functions

dZhei98 committed:
Change relevant classes to be abstract

Set Duke, InputManager, Parser, Task, TaskManager to be abstract classes

dZhei98 committed:
Add error handling for invalid input

dZhei98 committed:
Add error handling for invalid description

Handle invalid description for Todo, Deadline, and Event classes

dZhei98 committed:
Add error handling for missing task number

Also changed InvalidDescriptionException to MissingDescriptionException and update the respective exceptions accordingly

dZhei98 committed:
Add error handling for invalid task number

Add error handling for non-integer task number and out of bounds task number

dZhei98 committed:
Merge pull request #1 from joshuan98/branch-Level-5

Merge branch 'branch-Level-5'

dZhei98 committed:
Separate into packages

Create command, exception, task packages

dZhei98 committed:
Merge pull request #2 from joshuan98/branch-A-Packages

branch-A-Packages

dZhei98 committed:
Use ArrayList to store tasks

dZhei98 committed:
Add support for deleting tasks

dZhei98 committed:
Add support for deleting tasks

dZhei98 committed:
Add support to save and load tasks

Data stored in "ip/data/duke.txt"

dZhei98 committed:
Merge pull request #3 from joshuan98/branch-Level-7

Add support to save and load tasks

dZhei98 committed:
Update to use ArrayList for Level-7

dZhei98 committed:
Refactor writeFile in FileManager

dZhei98 committed:
Update .gitignore

dZhei98 committed:
Merge pull request #4 from joshuan98/branch-Level-6

Add support for deleting tasks

richwill28 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,49 @@
+        System.out.println(EXIT);
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println();
+    }
```
[Code Quality] The UI-related methods could probably be isolated to a dedicated UI class.

dZhei98 replied:
Isolated UI-related methods into Ui.java

richwill28 commented on src/main/java/duke/command/FileManager.java:
```diff
@@ -0,0 +1,96 @@
+            InputManager.handleInput(inputMark);
+        } catch (DukeException e) {
+            System.out.println("Bad file input :(");
+        }
```
[Code Quality] The `try-catch` clauses could be combined to one for better readability and control flow.

dZhei98 replied:
Combined relevant `try-catch` clauses

richwill28 commented on src/main/java/duke/command/InputManager.java:
```diff
@@ -0,0 +1,66 @@
+            TaskManager.deleteTask(input);
+        } else {
+            throw new DukeException();
+        }
```
Good SLAP!

richwill28 commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,259 @@
+            newTask = null;
+
+            break;
+        }
```
[Code Quality] This whole section is not very friendly to read. In general, we would simplify/isolate the logic from each case block. You could also consider how to better manage the exceptions handling to make the happy path more prominent.

dZhei98 replied:
Isolated logic into individual methods

richwill28 commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,259 @@
+                taskNumInt = Integer.parseInt(taskNumString);
+            } catch (NumberFormatException e) {
+                throw new NonIntegerTaskNumberException();
+            }
```
[Code Quality] Again, the `try-catch` can be combined.

dZhei98 replied:
Combined relevant `try-catch` statements

richwill28 reviewed (commented):
Overall, good work. For the next iterations you might want to consider how to improve the code architecture. Your code could also benefit with more documentation. Oh and please pay attention to the coding standards.

dZhei98 committed:
Extract Ui, storage, parser, tasklist classes

dZhei98 committed:
Use deadline dates as LocalDate objects

dZhei98 committed:
Find task by keyword

dZhei98 committed:
Add javadoc comments

Add to most non-public methods
Did not add comments for exceptions

dZhei98 committed:
Merge pull request #5 from joshuan98/branch-Level-8

Use deadline dates as LocalDate objects

dZhei98 committed:
Merge branch 'master' into branch-Level-9

dZhei98 committed:
Merge branch 'master' into branch-A-JavaDoc

dZhei98 committed:
Merge pull request #6 from joshuan98/branch-Level-9

Branch level 9

dZhei98 committed:
Merge branch 'master' into branch-A-JavaDoc

dZhei98 committed:
Merge pull request #7 from joshuan98/branch-A-JavaDoc

Add javadoc comments

dZhei98 committed:
Create _config.yml

dZhei98 committed:
Add user guide

dZhei98 committed:
Shift introduction and exit to Ui class

dZhei98 committed:
Elaborate on javadoc comments for task classes

dZhei98 committed:
Combine try-catch statements in parser and storage

Refactor some methods in parser to make happy path more prominent

dZhei98 committed:
Update file path

dZhei98 committed:
Create new directory and file if file not found

dZhei98 committed:
Update runtest.bat

dZhei98 committed:
Update readme to include setup
